rpcserver.go

```diff
@@ -1564,6 +1564,22 @@ func (r *rpcServer) OpenChannel(in *lnrpc.OpenChannelRequest,
 
 	localFundingAmt := btcutil.Amount(in.LocalFundingAmount)
 	remoteInitialBalance := btcutil.Amount(in.PushSat)
+
+	// If the FundAll flag is set, ensure that the
+	// acceptable minimum local amount adheres to the
+	// amount to be pushed to the remote, and to current
+	// rules.
+	if in.FundAll {
+		// Use as minmimum local amount the larger of
+		// the amount to be pushed to the remote or
+		// the allowed minimum funding size.
+		if remoteInitialBalance >= minChanFundingSize {
+			localFundingAmt = remoteInitialBalance + 1
```

Comment:
One requirement for local amount is to be greater than the remote initial balance. Is it therefore enough for the local amount to be 1 Satoshi greater?

Reply:
Where does this requirement come from?

Reply:
In practice, the remote amount cannot go over a certain percent because of the channel reserve. The same probably also applies to the push amount though I haven't tried this.
For this PR I think it either makes sense to disallow setting a push amount (`remoteInitialBalance`) in the command line when using `--fundall/--fundmax` or try to actually calculate the channel reserve in percent and then scale the local amount accordingly.

Reply:
I have tried to figure this out but am not completely sure. 

The way I see it, here in the code (https://github.com/lightningnetwork/lnd/blob/master/rpcserver.go#L1633-L1645) it "lazily" checks only if the push amount is greater or equal the local amount.

The logic to actually make sure the funds are sufficient happens later in the workflow here (https://github.com/lightningnetwork/lnd/blob/0dfb946afd36175dfc0e37acf45150784c87e375/lnwallet/reservation.go#L218-L260).

Therefore I was thinking to simply set the "minimum amount" (local_amt) to the greater of the existing channel opening minimum (`minChanFundingSize`) and the push amount, and let the existing logic later on make sure we actually have the funds to pay for fees, etc?

Reply:
Yes, you are right, there is a check later on. Found that out while testing :)
So I think that `remoteInitialBalance+1` is ok here.